app/src/backlight.c

```diff
@@ -114,11 +96,31 @@ static int zmk_backlight_init(const struct device *_arg) {
     return zmk_backlight_update();
 }
 
-int zmk_backlight_set_on(bool on) {
+static int zmk_backlight_save_state() {
+#if IS_ENABLED(CONFIG_SETTINGS)
+    k_delayed_work_cancel(&backlight_save_work);
+    return k_delayed_work_submit(&backlight_save_work, K_MSEC(CONFIG_ZMK_SETTINGS_SAVE_DEBOUNCE));
+#else
+    return 0;
+#endif
+}
+
+int zmk_backlight_get_state(bool *on_off) {
+    if (!backlight_dev)
+        return -ENODEV;
```

Comment:
`backlight_dev` is always not `NULL`, this check is useless.

Reply:
This is if backlight behaviours are called without backlight being defined

Reply:
Using a split however you could have it properly defined on one side calling the function on the other

Reply:
I think this shouldn't be allowed and I don't even think it will work as it is not possible to enable `CONFIG_ZMK_BACKLIGHT` without a device

Reply:
but you can include the behaviours file and have the central attempt to call the function on the peripheral. The ability to ensure that edge cases get caught seems like a worthwhile addition? I can remove it if you're sure it's superfluous

Reply:
You can include the behavior header, however, unless you enable `CONFIG_ZMK_BACKLIGHT`, any presses will be ignored.

Reply:
Ok i see, will remove